os/arch/arm/src/imxrt/imxrt_pwm.c

```diff
@@ -874,11 +890,39 @@ static int imxrt_pwm_setup(FAR struct pwm_lowerhalf_s *dev)
  *
  ****************************************************************************/
 static int imxrt_pwm_start(FAR struct pwm_lowerhalf_s *dev,
-						 FAR const struct pwm_info_s *info)
+						   FAR const struct pwm_info_s *info)
 {
 	FAR struct imxrt_pwmtimer_s *priv = (FAR struct imxrt_pwmtimer_s *)dev;
 
-	imxrt_pwm_starttimer(priv->base, kPWM_Control_Module_0 | kPWM_Control_Module_1 | kPWM_Control_Module_2);
+	imxrt_pwm_setupforcesignal(priv->base, kPWM_Control_Module_0, kPWM_PwmA, kPWM_UsePwm);
+	imxrt_pwm_setupforcesignal(priv->base, kPWM_Control_Module_0, kPWM_PwmB, kPWM_UsePwm);
+	imxrt_pwm_setupforcesignal(priv->base, kPWM_Control_Module_1, kPWM_PwmA, kPWM_UsePwm);
+	imxrt_pwm_setupforcesignal(priv->base, kPWM_Control_Module_1, kPWM_PwmB, kPWM_UsePwm);
+	imxrt_pwm_setupforcesignal(priv->base, kPWM_Control_Module_2, kPWM_PwmA, kPWM_UsePwm);
+	imxrt_pwm_setupforcesignal(priv->base, kPWM_Control_Module_2, kPWM_PwmB, kPWM_UsePwm);
+
+	priv->base->SM[kPWM_Module_0].CTRL2 |= PWM_CTRL2_FORCE(1);
+	priv->base->SM[kPWM_Module_1].CTRL2 |= PWM_CTRL2_FORCE(1);
+	priv->base->SM[kPWM_Module_2].CTRL2 |= PWM_CTRL2_FORCE(1);
+
+	printf("info->duty: %d, info->frequency: %d\n", info->duty, info->frequency);
+
+	if (priv->frequency != info->frequency || priv->duty != info->duty) {
+		imxrt_pwm_stoptimer(priv->base, kPWM_Control_Module_0 | kPWM_Control_Module_1 | kPWM_Control_Module_2);
+
+		for (int i = 0; i < IMXRT_PWM_CNT; i++) {
+			imxrt_pwm_drv_init3phpwm(i, info->duty, info->frequency);
```

Comment:
Logic is totally wrong.
You are controlling another path's dev here, in other words you are modifying another iNode without each iNode's file operation here.
Please make sure that you have to control each of iNode with each of fd.
Please assume that some threads using each of fd from each of diffrent path.....